specs/vep-554-apache-airflow-integration/README.MD

```diff
@@ -0,0 +1,401 @@
+  start a data job execution and wait for its completion or just start data job execution
+  and return an execution id. The `asynchronous`=True is used in combination with
+  VDKSensorOperator (more on that below).
+* on_kill - This method cancels the Data Job execution when an Airflow task instance gets
+  killed.
+
+Usage:
+
+```python
+from airflow import DAG
+from airflow.providers.vdk.operators.vdk import VDKOperator
+
+with DAG:
+   # [START sync_data_job]
+   sync_data_job = VDKOperator(
+      job_name='sync_data_job',
+   )
+   # [END sync_data_job]
+
+   sync_data_job
+```
+
+#### VDK Sensor Operator
+
+Airflow provides a primitive for a special kind of operator, whose purpose is to poll
+some state (e.g. data job completion) at regular intervals until a success criterion is
+met. Sensors have a powerful feature called 'reschedule' mode, which allows the Airflow
+Scheduler to reschedule a sensor task, rather than blocking a worker slot between pokes.
+This is useful when you can tolerate a longer poll interval and expect to be polling for
+a long time.
+
+VDKSensorOperator is responsible for the **asynchronous** execution of Data Jobs in
+combination with VDKOperator. As the sensors only have the ability to poll the state of
+data job execution, the VDKOperator has to be used to start it. The sensor is defined by
+extending the airflow.sensors.base.BaseSensorOperator and defining a `poke` method to
+poll Data Job execution state and evaluate the job execution completion.
+
+Usage:
+
+```python
+from airflow import DAG
+from airflow.providers.vdk.operators.vdk import VDKOperator
+from airflow.providers.vdk.sensors.vdk import VDKSensorOperator
+
+with DAG:
+   # [START async_data_job]
+   start_async_data_job = VDKOperator(
+      job_name='async_data_job',
+      asynchronous=True,
+   )
+
+   async_data_job = VDKSensorOperator(
+      job_name='async_data_job',
+      job_execution_id=start_async_data_job.output,
+   )
+   # [END async_data_job]
+
+   async_data_job
+```
+
+#### Sample DAG
+
+```python
+from airflow import DAG
+from airflow.providers.vdk.operators.vdk import VDKOperator
+from airflow.providers.vdk.sensors.vdk import VDKSensorOperator
+
+with DAG(
+   dag_id='sample_vdk_dag',
+   schedule_interval=None,
+   start_date=datetime(2022, 3, 3),
+   catchup=False,
+   default_args={'team_name': 'default_team_name', 'conn_id': 'default_conn_id'},
+) as dag:
+   # [START sync_data_job]
+   sync_data_job = VDKOperator(
+      job_name='sync_data_job',
+   )
+   # [END sync_data_job]
+
+   # [START async_data_job]
+   start_async_data_job = VDKOperator(
+      job_name='async_data_job',
+      asynchronous=True,
+   )
+
+   async_data_job = VDKSensorOperator(
+      job_name='async_data_job',
+      job_execution_id=start_async_data_job.output,
+   )
+   # [END async_data_job]
+
+   sync_data_job >> async_data_job
+```
+
+## Security and Permissions
```

Comment:
In the latest template (on advice of Rumen) I put this under "Detailed design" so you may add one extra "#"